src/algorithms/simulation/circuit_to_truthtable.cpp

```diff
@@ -16,13 +20,27 @@ namespace syrec {
 
         while (n < totalInputs) {
             const auto inCube = TruthTable::Cube::fromInteger(n, nBits);
+            ++n;
+
+            const auto boolCube  = inCube.toBoolVec();
+            bool       nextInput = false;
 
-            auto const inEdge    = dd->makeBasisState(nBits, inCube.toBoolVec());
+            for (auto i = 0U; i < nBits; i++) {
+                if (qc.getAncillary()[i] && (boolCube[i])) {
```

Comment:
shouldn't this already be using the `tt` constants? at least that way the getter functions would not be unused.
the truth table class could benefit from a `isConstant(std::size_t i) -> bool` function (and something similar for the garbage).
In the same sense `setConstant(std::size_t i) -> void` could be helpful.